### ISSUE TITLE
CMake/Embree issues on VS

### ISSUE THREAD

**Issue by alecjacobson** [libigl/libigl#219]
From personal email:

> tried to use libigl by adding the the required libs to my C++ project and managed to debug the project without any issue. 
BUT, later I tried to run the tutorials in VS 2013. So I used CMake to generate the visual studio solutions of the tutorials in tutorial folder and ran into some issues like  "could not find embree" .  In below, I have attached an screen shot of the errors in CMake . It would be great if you could help me to solve this issue! 
![image](https://cloud.githubusercontent.com/assets/2241689/12678262/f496962e-c66b-11e5-9529-d65ad24f2fde.png)

**Comment by alecjacobson** [comment]
It's possible this is a visual studio problem, can you upgrade to VS 2015 community edition?

**Comment by ArashPix** [comment]
I guess it is not related to VS version. I have already tested in VS 2010 and 2013!!!
Shall I install CGAL before building the tutorials with CMake?

**Comment by danielepanozzo** [comment]
We did not test it with older VS versions, could you try to build all the tutorials (use the cmake file in the tutorial folder, not the individual examples cmake files) using VS 2015 community edition and the latest version of cmake? To reduce the chances of other issues due to your changes, could you also re-clone the repository and make sure you used the --recursive option?

**Comment by danielepanozzo** [comment]
Another minor thing: try to remove the spaces (" ") in your path, they sometime create issues in cmake if they are not carefully handled everywhere.

**Comment by ArashPix** [comment]
@danielepanozzo:
Thanks for your reply! finally one of my colleague helped me to build the code by modifying the CMake file. Now, tutorials can be run on VS 2013. If anyone has the same issue I can send him our CMake file. 

By the way, I have one last question. I've seen in the demos of your papers in SIGGRAPH, the software application has a UI system which contains some UI elements like buttons and sliders. Have you included the UI as well in the library?

**Comment by danielepanozzo** [comment]
Could you send a pull request with the changes? If they do not break the VS2015 build I would like to include them. 

It is an external library called nanogui, you can see an example in http://libigl.github.io/libigl/tutorial/tutorial.html#viewermenu

**Comment by ArashPix** [comment]
Hey, sorry for my late reply. We only have modified a few lines in CMakeLists file in tutorial folder. So here, I have uploaded the new CMakeLists.txt. Building the tutorials through this CMakeLists work well for both VS 2013 and VS 2015. I have also managed to display GUI(nanogui) for all examples(required changes are addressed in the new CMake file).
[CMakeLists.txt](https://github.com/libigl/libigl/files/122849/CMakeLists.txt)
